### PR TITLE
docs: RFC proposal to center talk about authentication around

### PR DESCRIPTION
**Pull request by bcoe**
I don't have particularly strong opinions about how authentication is implemented in yarn, but I've had a few customers ask after it already, and there are a few threads in the yarn issue tracker on the subject (https://github.com/yarnpkg/yarn/issues/521, https://github.com/yarnpkg/yarn/issues/547).

I thought it would be worthwhile to center this discussion around an RFC; it's a complicated enough feature that context could easily be lost across open GitHub issues.

### CODE REVIEW

**Comment by joshmanders** [comment]
Yes! I would like to see something that supports context switching. I have a tool that I wrap around npm right now that allows me to login to multiple accounts and switch out the authentication token depending on a `--as` flag. Since I publish modules under my own account and employers account, sometimes it gets cumbersome logging out and into new accounts to make the changes, then back.

**Comment by ljharb** [comment]
I do the same with an alias that moves around .npmrc files. It'd be great to have yarn more transparently use npm configs as opposed to inventing a non-drop-in way to configure things.

**Comment by ashleygwilliams** [comment]
relevant: https://github.com/yarnpkg/yarn/pull/839

tl;dr this PR implements using a `.npmrc` file to auth to private registry

**Comment by bcoe** [comment]
@ashleygwilliams yarnpkg/yarn#839 seems like a great approach to implementing authentication; Supporting multiple registries is definitely the best-case scenario.

**Comment by bcoe** [comment]
It looks like @devongovett has pulled together a wonderful implementation of authentication/multiple-registry-support. Closing this RFC, and calling it a fun experiment 👍
